src/main/java/org/stellar/sdk/MemoHashAbstract.java

```diff
@@ -54,7 +54,11 @@ public String getHexValue() {
    * </code>
    */
   public String getTrimmedHexValue() {
-    return this.getHexValue().split("00")[0];
+      String[] strArray = this.getHexValue().split("00");
+      if (strArray.length > 0) {
+          return strArray[0];
```

Comment:
I think the bug here is that we attempt to trim the hash value at the first 0x00. Memo hashes are always 32bytes, fixed size, and so we should always return the full 32byte hash without truncating. By trimming at the first 0x00 we might be losing non-zero bytes that follow.

Reply:
I think what you're saying, @leighmcculloch, is that `getTrimmedHexValue()` shouldn't exist, because the very notion of trimming a memo hash is incorrect (see docstring). There must be a historical reason for this helper, then :thinking:

Reply:
Or, alternatively, this should do a `stripEnd("00")`? Not sure if that [usage](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#stripEnd-java.lang.String-java.lang.String-) is correct.

Reply:
Yes we should remove getTrimmedHexValue. If an application is using a memo hash in this way, they should be using a different type of memo, or doing the truncation in their application. There is really no way that the Java SDK can know if a zero is safe to truncate, because zero is a significant data point in a hash.

Reply:
+1 Applications can use the hash as a null terminated string if they choose, but that logic should live in the application, not in the SDK.